propagator/jaeger/lib/opentelemetry/propagator/jaeger/text_map_extractor.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# OpenTelemetry is an open source observability framework, providing a
+# general-purpose API, SDK, and related tools required for the instrumentation
+# of cloud-native software, frameworks, and libraries.
+#
+# The OpenTelemetry module provides global accessors for telemetry objects.
+# See the documentation for the `opentelemetry-api` gem for details.
+module OpenTelemetry
+  # Namespace for OpenTelemetry propagator extension libraries
+  module Propagator
+    # Namespace for OpenTelemetry Jaeger propagation
+    module Jaeger
+      # Extracts context from carriers
+      class TextMapExtractor
+        TRACE_SPAN_IDENTITY_REGEX = /\A(?<trace_id>(?:[0-9a-f]{16}){1,2}):(?<span_id>[0-9a-f]{16}):[0-9a-f]{1,16}:(?<sampling_flags>[0-9a-f]{1,2})\z/.freeze
+        SAMPLED_VALUES = %w[1 3 5 7].freeze
```

Comment:
You shouldn't need to enumerate these values. Just parse the flags, convert to an integer, and use a bitwise & to test for sampled or debug.